main.tf

```diff
@@ -16,36 +16,66 @@
 
 
 locals {
-  address                 = var.create_address ? join("", google_compute_global_address.default.*.address) : var.address
+  ipv4_address            = var.create_ipv4_address ? join("", google_compute_global_address.default_ipv4.*.address) : var.ipv4_address
+  ipv6_address            = var.create_ipv6_address ? join("", google_compute_global_address.default_ipv6.*.address) : var.ipv6_address
   url_map                 = var.create_url_map ? join("", google_compute_url_map.default.*.self_link) : var.url_map
   create_http_forward     = var.http_forward || var.https_redirect
   health_checked_backends = { for backend_index, backend_value in var.backends : backend_index => backend_value if backend_value["health_check"] != null }
 }
 
-resource "google_compute_global_forwarding_rule" "http" {
+### IPv4 block ###
+resource "google_compute_global_forwarding_rule" "http_ipv4" {
```

Comment:
Agreed. coming up in next PR.

Reply:
But, do plan out refactoring the names to make them more descriptive and relatable.

Reply:
It would be best if we didn't change the names actually as I don't want to force existing users to recreate their addresses.

Reply:
Sure, your call.

I just thought it would be more readable to have explicit and unambiguous names like `create_ipv4_address` instead of `create_address`.

Maybe, via a pre-announced deprecation/migration plan? - just a thought.

Reply:
To be clear, I'm fine with the proposed *variable* names (as those are an easy fix to make in-place). I just don't want to change the *resource* name as it will cause address recreation.

Reply:
Ah, ok.

For now, I'm not touching the existing variable names as well - just to be on safe side.
We can plan that out for future some time.